drv/lpc55-rng/src/main.rs

```diff
@@ -197,14 +224,95 @@ impl NotificationHandler for Lpc55RngServer {
     }
 }
 
+/// Load a type implementing HandoffData (and others) from a config::DataRegion.
+/// Errors will be reported in the ringbuf and will return in None.
+#[cfg(feature = "dice-seed")]
+fn load_data_from_region<
+    T: for<'a> Deserialize<'a> + HandoffData + SerializedSize,
+>(
+    region: &DataRegion,
+) -> Option<T> {
+    use core::slice;
+
+    // Safety: This memory is setup by code executed before hubris and
+    // exposed using the kernel `extern-regions` mechanism. The safety of
+    // this code is an extension of our trust in the hubris kernel / build.
+    let data = unsafe {
+        slice::from_raw_parts(region.address as *mut u8, region.size as usize)
```

Comment:
**question:** Please excuse my dumb question / confirmation, but I presume "this memory" is loaded into actual memory, and is not MMIO or something like that? In which case it might not be safe to turn it into a slice for reading.